### PR TITLE
Fix for thimble.mozilla.org#2237

### PR DESCRIPTION
**Pull request by Cryzek**
Fixed mozilla/thimble.mozilla.org#2237 - "After deleting a file, it's name is still indicated in the Editor header bar."
Earlier, when last file was deleted, it was returning from the function without calling on the editor change event.
Right now, I have added a workaround for this, by sending an object with filename as "No File" which gets set in editor-nav-pane.

### CODE REVIEW

File: src/extensions/default/bramble/lib/RemoteEvents.js [hunk 1]
```diff
@@ -98,6 +98,13 @@ define(function (require, exports, module) {
         var lastKnownEditorFilePath;
         MainViewManager.on("currentFileChange", function(e, file) {
             if(!file) {
+                // This happens when the last file in working set is deleted.
+                lastKnownEditorFilePath = "";
+                var noFile = {
+                    fullPath: "No File",
+                    filename: "No File"
```
Comment (humphd) [inline src/extensions/default/bramble/lib/RemoteEvents.js]: I think this is good. Question: do we care about l10n here?  This is API level, not user, so we can deal this in Thimble.  But if so, we need to do it, and I bet we are currently just displaying whatever `filename` is.  Also, I think we cache `fullPath` to try and re-open it on the next startup.  We should figure out what to do here.

Comment (Cryzek) [reply src/extensions/default/bramble/lib/RemoteEvents.js]: Thanks for the review.
I think l10n work can be done. 
But, I don't understand how to start with the caching issue you mentioned.

Comment (humphd) [reply src/extensions/default/bramble/lib/RemoteEvents.js]: The Brackets iframe API that runs within Thimble, and listens for this event, is here:

https://github.com/mozilla/brackets/blob/master/src/bramble/client/main.js#L422-L424

When we start the editor, we do some work here to figure out which file to load, checking for that cached path:

https://github.com/mozilla/brackets/blob/master/src/bramble/client/main.js#L524

Probably if you store `null` vs. `"No File"` it should do we want, but I'd have to research a bit more.

**Comment by flukeout** [comment]
Testing this out now.

**Comment by flukeout** [comment]
Cool just tested this now. It correctly displays ``No File`` when there isn't anything left to open or edit. However, I think we can just ditch the ``No File`` heading. Just removing the previous filename is enough to solve this. Maybe that helps with some of the issues you are discussing above?

Thanks for your work on this @Cryzek!

**Comment by Cryzek** [comment]
@flukeout Thanks for the suggestion. Displaying nothing in editor header ,certainly takes away the l10n part.

@humphd I did not notice any issues, on reloading once all files were deleted. A new index.html was automatically inserted on the next load. 

Also, I stumbled upon https://github.com/mozilla/brackets/blob/master/src/view/Pane.js#L1045-L1065 which does the same thing but in some other part of brackets, some ".pane-header". It displays filename relative to the project root. 
For eg. for files inside folders other than root folder, only the home.html is displayed.
capture

Should "Tests/home.html" be displayed instead?
If so, should I try it in this issue or a new issue should be opened for it?

**Comment by humphd** [comment]
Interesting.  @flukeout do you want the project-relative path to be displayed, or just the filename itself?

**Comment by flukeout** [comment]
Hey good catch @Cryzek - I think it's OK just to indicate the filename. Often there isn't a ton of room there anyway so adding a folder path might make it too long.

**Comment by flukeout** [comment]
Good stuff @Cryzek - I'll re-test now.

**Comment by flukeout** [comment]
Cool yeah this is working great. No filename is displayed when the editor doesn't have a file open.

Good to go from my end as long as the code review is 👍 @humphd @Cryzek

**Comment by flukeout** [comment]
Ping @humphd - can you check into this one? It's good to go pending a code review.

**Comment by humphd** [comment]
Sorry for the delay on this, this looks good to me.  Merging.  Thank you @Cryzek!

**Pull request merged**